### ISSUE TITLE
Number of packages

### ISSUE THREAD

**Issue by Kapeli** [mofeing/julia-package-docsets#2]
The User Contributed repo is not really designed to handle a huge amount of packages, so please note that if the number of packages in this repo will increase by a lot, we'll have to remove them from user contributed and find a different place for them.

**Comment by mofeing** [comment]
Oh, okay. My idea was to extend it a lot more... 😥

mmm can we instead extend Dash such that it fetches the documentation from JuliaHub (de-facto frontend to the official registry)? Something like the Java, Scala, Ruby... docsets. 

The scrapper and parser from this repo could be reused. The only missing part would be the search engine.

(Example of a search in JuliaHub: https://juliahub.com/ui/Search?q=chainrules&type=packages)

**Comment by Kapeli** [comment]
I see. Hmm. I'll remove the packages that were added and figure something else out.

To add a docset repo to Dash I need the following:
1. An API to search for packages
2. An API to get the versions available for packages
3. A way to get the (offline HTML) docs given a package and version

Any ideas on how to get these for Julia packages?

**Comment by mofeing** [comment]
I'm pinging @Dattax as he is probably the person from inside JuliaHub that we should contact for this matters. he is proposing a call between the 3 of us next week. is it ok for you @Kapeli?

**Comment by Kapeli** [comment]
Unfortunately I'm not able to schedule calls considering the recent release of Dash 7. If possible, can we do this over email or right here?

**Comment by Dattax** [comment]
@Kapeli - I'm investigating on access to those API asks and will get back to you both next week.

**Comment by mofeing** [comment]
Yes, we can continue through here. I spoke to @Dattax and he is going to ask internally in JuliaHub. The API needs are mostly fulfilled through the search engine on the web, but the API might be private right now.

Anyway, @Dattax will speak with the appropriate people and contact us.

**Comment by Dattax** [comment]
We don't have public APIs that will allow you to do this right now. We could develop them, but our roadmap is pretty impacted so I'm not sure how fast you need this or how big a priority it is.

**Comment by mofeing** [comment]
@Kapeli would it be possible to parse the results from the JuliaHub search engine instead? At least temporarily until the APIs are available.

**Comment by mofeing** [comment]
Btw, some information about the documentation generation process in Julia. Docs generated by [Documenter.jl](https://juliacollections.github.io/AbstractTrees.jl/versions.js) (kind of like the defacto docs generator) provide a `versions.js` file that points to the dir containing each versioned docs.

For example, the content of https://juliacollections.github.io/AbstractTrees.jl/versions.js is:
```js
var DOC_VERSIONS = [
  "stable",
  "v0.4",
  "v0.3",
  "dev",
];
```

And thus, there exists the following versioned docs:
- https://juliacollections.github.io/AbstractTrees.jl/v0.4
- https://juliacollections.github.io/AbstractTrees.jl/v0.3
- https://juliacollections.github.io/AbstractTrees.jl/dev which is _usually_ generated on each push to the `master` branch
- https://juliacollections.github.io/AbstractTrees.jl/stable which points to the latest version (i.e. v0.4)

This file could be used for getting versioned docs, but the tricky part is to get the URL to the docs given a package.

Keeping the docs organized and hosting them (although usually GitHub Pages is used for that) is the responsibility of the package developer. Although the general rule is to have docs, there are packages which do not provide them.

From what I see, JuliaHub generates the docs on their own **just for the last released version** of the packages and also for packages which do not have docs configured (correct me if I'm wrong @Dattax).

**Comment by Kapeli** [comment]
> @Kapeli would it be possible to parse the results from the JuliaHub search engine instead? At least temporarily until the APIs are available.

Parsing/scraping is always an option, but I like to build things with APIs for the third-party sources, so they don't break easily.

@Dattax Where should I open a GitHub issue to request these APIs to be added? I can't really appreciate the priority or urgency. Dash will make use of the APIs whenever they're available.

**Comment by Dattax** [comment]
You can open issues at anytime here: https://github.com/JuliaComputing/JuliaHub/issues

I'll look out for the issue :)

**Comment by Kapeli** [comment]
> You can open issues at anytime here: https://github.com/JuliaComputing/JuliaHub/issues

This link points to a page that doesn't exist.

**Comment by Dattax** [comment]
Ah sorry - try this one: https://github.com/JuliaComputing/JuliaHub.jl/issues
